src/main/java/org/hawkular/client/android/util/Preferences.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.util;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+
+import info.metadude.android.typedpreferences.StringPreference;
+
+public final class Preferences {
+    private static final class Locations {
+        private Locations() {
+        }
+
+        public static final String BACKEND = "backend";
+    }
+
+    public static final class Keys {
+        private Keys() {
+        }
+
+        public static final String BACKEND_HOST = "host";
+        public static final String BACKEND_PORT = "port";
+        public static final String BACKEND_TENANT = "tenant";
+        public static final String BACKEND_ENVIRONMENT = "environment";
+    }
+
+    private SharedPreferences serverPreferences;
+
+    @NonNull
+    public static Preferences ofBackend(@NonNull Context context) {
```

Comment:
Let's change it to `of(@NonNull Context context)` to make it concise with the other builders/fluent interfaces